### DISCUSSION TITLE
Property IsAvailableforMDX set to false, Cardinality goes to 0?

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1239] [🙋♀️TE3 Q&A] [answered]
How is it possible if you set Property IsAvailableforMDX set to false, Cardinality goes to 0?
I thought just the hierarchy would drop in size.

How does anything work without cardinaltiy.  If I set Property IsAvailableforMDX set to false on something customer region.  I can still get total orders by customer region.

**Comment by edhans** [comment]
Cardinality doesn't change. Just the metrics that Vertipaq Analyzer uses. 

In Vertipaq Analyzer settings (preferences) turn on "Read data from statistics" and see if you get results. It will take longer but cardinality will return. It has to use other tools to do it, it cannot use the data in the hierarchies. 

Note that "read data from stats" can cause other issues, so 99% of the time I turn it back off. I am not sure why. I never have an issue with Vertipaq Analyzer in DAX Studio, and I don't understand how DS and TE/3 have implemented Vertipaq Analyzer differently.

**Reply by TinyTimothy2** [reply]
The issues are important to know.  You have to be very careful to turn it back off.

**Comment by TinyTimothy2** [comment]
So this is a bug with TE3?  because in TE3 cardinality goes to ZERO and it will stay at zero no matter how long you leave it open

**Reply by otykier** [reply]
No - the problem is that if you disable attribute hierarchies (which is what happens when you set IsAvailableInMdx to false), VertiPaq Analyzer won’t be able to obtain the cardinality of the column unless “Read statistics from data” is enabled. You would see the same thing in DAX Studio or if using the standalone VertiPaq Analyzer tool.

The “Read statistics from data” option forces VertiPaq Analyzer to execute suitable DAX queries (i.e. `DISTINCTCOUNT([Column])`) to determine the cardinality. It works, but it’s slower than reading it from the statistics DMVs.

**Comment by TinyTimothy2** [comment]
Thank you.  Where do you find the options The “Read statistics from data” option in TE3?
This is also giving me second thoughts on setting to FALSE on FK.  I have not measured any timings, but don't observe any changes when running report with DISTINCTCOUNT([FK])

(Please note this is the first opportunity i have had to explore this option)

**Reply by edhans** [reply, marked as answer]
Go to preferences, Vertipaq Analyzer, check "read stats from data"

**Comment by TinyTimothy2** [comment]
Thank you for the responses.  if you use 'IsAvailableforMDX set to false' it seems you must check preferences, Vertipaq Analyzer, check "read stats from data".
A nice enhancement would be a check if users have it set and alert to the pref change.
A timing would also be nice to see how much longer it takes to run.
To be cardinality is # 1 item I look at, so i have change the pref.
